drivers/video/video_mcux_mipi_csi2rx.c

```diff
@@ -0,0 +1,215 @@
+			720,
+			30,
+			0x12,
+		},
+		{
+			1280,
+			720,
+			15,
+			0x17,
+		},
+		{
+			640,
+			480,
+			30,
+			0x1F,
+		},
+		{
+			640,
+			480,
+			15,
+			0x24,
+		},
+		{
+			320,
+			240,
+			30,
+			0x1F,
+		},
+		{
+			320,
+			240,
+			15,
+			0x24,
+		},
+	};
+
+	csi2rxConfig.laneNum = DEFAULT_MIPI_CSI_NUM_LANES;
+
+	for (i = 0; i < ARRAY_SIZE(csi2rxHsSettle); i++) {
+		if ((fmt->width == csi2rxHsSettle[i][0]) && (fmt->height == csi2rxHsSettle[i][1]) &&
+		    (DEFAULT_CAMERA_FRAME_RATE == csi2rxHsSettle[i][2])) {
+			csi2rxConfig.tHsSettle_EscClk = csi2rxHsSettle[i][3];
+			break;
+		}
+	}
+
+	if (i == ARRAY_SIZE(csi2rxHsSettle)) {
+		LOG_ERR("Unsupported resolution");
+		return -ENOTSUP;
+	}
+
+	drv_data->csi2rxConfig = csi2rxConfig;
+
+	if (video_set_format(config->sensor_dev, ep, fmt)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int mipi_csi2rx_get_fmt(const struct device *dev, enum video_endpoint_id ep,
+			       struct video_format *fmt)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+
+	if (fmt == NULL || ep != VIDEO_EP_OUT) {
+		return -EINVAL;
+	}
+
+	if (video_get_format(config->sensor_dev, ep, fmt)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int mipi_csi2rx_stream_start(const struct device *dev)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+	struct mipi_csi2rx_data *drv_data = dev->data;
+
+	CSI2RX_Init((MIPI_CSI2RX_Type *)config->base, &drv_data->csi2rxConfig);
+
+	if (video_stream_start(config->sensor_dev)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int mipi_csi2rx_stream_stop(const struct device *dev)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+
+	if (video_stream_stop(config->sensor_dev)) {
+		return -EIO;
+	}
+
+	CSI2RX_Deinit((MIPI_CSI2RX_Type *)config->base);
+
+	return 0;
+}
+
+static int mipi_csi2rx_get_caps(const struct device *dev, enum video_endpoint_id ep,
+				struct video_caps *caps)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+
+	if (ep != VIDEO_EP_OUT) {
+		return -EINVAL;
+	}
+
+	/* Just forward to sensor dev for now */
+	return video_get_caps(config->sensor_dev, ep, caps);
+}
+
+static const struct video_driver_api mipi_csi2rx_driver_api = {
+	.get_caps = mipi_csi2rx_get_caps,
+	.get_format = mipi_csi2rx_get_fmt,
+	.set_format = mipi_csi2rx_set_fmt,
+	.stream_start = mipi_csi2rx_stream_start,
+	.stream_stop = mipi_csi2rx_stream_stop,
+};
+
+static const struct mipi_csi2rx_config mipi_csi2rx_config_0 = {
+	.base = (MIPI_CSI2RX_Type *)DT_INST_REG_ADDR(0),
+	.sensor_dev = DEVICE_DT_GET(DT_INST_PHANDLE(0, sensor)),
+};
+
+static struct mipi_csi2rx_data mipi_csi2rx_data_0;
+
+static int mipi_csi2rx_init_0(const struct device *dev)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+
+	/* Check if there is any sensor device */
+	if (!device_is_ready(config->sensor_dev)) {
+		return -ENODEV;
+	}
+
+	/* MIPI DPHY power on and isolation off */
+	PGMC_BPC4->BPC_POWER_CTRL |= (PGMC_BPC_BPC_POWER_CTRL_PSW_ON_SOFT_MASK |
```

Comment:
This seems like it is probably specific to the RT11xx. We should either:
- guard this with `CONFIG_SOC_SERIES_IMXRT11XX`
- add a function like [imxrt_pre_init_display_interface](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nxp/imxrt/imxrt5xx/cm33/soc.h#L80) for the MIPI CSI, and implement it for the RT11XX

Reply:
MIPI CSI-2 Rx is an IP specific to RT11xx (RT10xx doesn't have this IP). So, I think it's ok to set this register here, in the driver. It's not the same case as the display (eLCDIF is common IP for both 10xx and 11xx)

Reply:
> MIPI CSI-2 Rx is an IP specific to RT11xx (RT10xx doesn't have this IP). So, I think it's ok to set this register here, in the driver. It's not the same case as the display (eLCDIF is common IP for both 10xx and 11xx)

It is specific *now*. There is no promise that future SOCs will not integrate this IP, and at that point code like this within the driver layer becomes an issue.

Reply:
Okay, I will guard it with "CONFIG_SOC_SERIES_IMXRT11XX".